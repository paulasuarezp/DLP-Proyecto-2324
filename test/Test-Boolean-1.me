class
	prog2;
	
global

	vars
		b : INTEGER;

	create
		inicio;
	feature inicio() is 
		local
			k : BOOLEAN;
			
		do
			k := true;
			b := false = 3 > 2;
			println b;
			b := true = 3 > 2;
			println true = false;
			println not k;
			println k and false;
			print k = b;
	end
end

run inicio(); 
