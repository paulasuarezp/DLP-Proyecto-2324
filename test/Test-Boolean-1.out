#source "input.txt"

'Clase: prog2
'Declaraciones globales
#global b : int

'**builder inicio

CALL inicio
HALT

#FUNC inicio
#ret void
inicio:
#local k : int
ENTER 2

#line 16
PUSH BP
PUSH -2
ADDI
PUSHI 1
STOREI

#line 17
PUSHA 0
PUSHI 0
PUSHI 3
PUSHI 2
GTI
EQI
STOREI

#LINE 18
PUSHA 0
LOADI
OUTI
PUSHB 10
OUTB

#line 19
PUSHA 0
PUSHI 1
PUSHI 3
PUSHI 2
GTI
EQI
STOREI

#LINE 20
PUSHI 1
PUSHI 0
EQI
OUTI
PUSHB 10
OUTB

#LINE 21
PUSH BP
PUSH -2
ADDI
LOADI
NOT
OUTI
PUSHB 10
OUTB

#LINE 22
PUSH BP
PUSH -2
ADDI
LOADI
PUSHI 0
AND
OUTI
PUSHB 10
OUTB

#LINE 23
PUSH BP
PUSH -2
ADDI
LOADI
PUSHA 0
LOADI
EQI
OUTI
RET 0, 2, 0
